include/osquery/core.h

```diff
@@ -15,8 +15,9 @@
 #include <string>
 #include <vector>
 
-#if defined(__APPLE__) || defined(__FreeBSD__)
+#if defined(__APPLE__) || defined(__FreeBSD__) || defined(__linux__)
```

Comment:
On Linux our build should link a C++14 compatible implementation statically, so we have a shared and recursive mutex.